src/validation.cpp

```diff
@@ -4091,12 +4091,11 @@ void CChainState::ProcessProposalEvents(const CBlockIndex* pindex, CCustomCSView
             return true;
         }, CMnVotePerCycle{propId, prop.cycle});
 
-        // Redistributes CFP fee among voting masternodes
-        CDataStructureV0 feeRedistributionKey{AttributeTypes::Governance, GovernanceIDs::CFP, GovernanceKeys::CFPFeeRedistribution};
-        if (prop.type == CPropType::CommunityFundProposal && voters.size() > 0 && attributes->GetValue(feeRedistributionKey, false)) {
-            auto proposalFee = GetPropsCreationFee(pindex->nHeight, prop);
+        // Redistributes fee among voting masternodes
+        CDataStructureV0 feeRedistributionKey{AttributeTypes::Governance, GovernanceIDs::Proposals, GovernanceKeys::FeeRedistribution};
+        if (voters.size() > 0 && attributes->GetValue(feeRedistributionKey, false)) {
             // return half fee among voting masternodes, the rest is burned at creation
-            auto feeBack = proposalFee / 2;
+            auto feeBack = prop.fee / 2;
```

Comment:
@Bushstar notice for the review that I changed getting the fee to use the fee which was at the time of creation, because it can change in the meantime.